src/tasks/group/taskgroup.cpp

```diff
@@ -51,9 +51,35 @@ void TaskGroup::cancel() {
   }
 }
 
-bool TaskGroup::deletable() const {
+Task::DeletePolicy TaskGroup::deletePolicy() const {
+  // The policy for task group is:
+  // - if there is at least 1 non-deletable task, the group is non-deletable
+  // - if the tasks are all reschedulable, the group is reschedulable
+  // - if there are no reschedulable tasks, the group is deletable
+  // - fallback: the task is not deletable
```

Comment:
Hum, it is not obvious to me why you landed on this policy for groups. Would you mind expanding?

Reply:
My idea is that the group should not override the policy of the sub-tasks.
1. The first point seems reasonable: if there is at least 1 non-deletable task the whole group is non-deletable.
2. The second one makes sense too: if all are reschedulable, the group is reschedulable.
3. The third one seems correct: if all are deletable -> let's delete the entire group
4. the last one is the complex one. If we are here, we have some reschedulable tasks, and some deletable ones. We don't know if changing the order of the execution we can introduce bugs, so my approach is: do nothing, keep the group as it is.

Reply:
```suggestion
  // - if there is at least 1 reschedulable task, the group in non-deletable
```
Just to make it clear.